flycheck.el

```diff
@@ -4609,6 +4609,49 @@ symbols in the command."
          (seq-map (lambda (arg) (flycheck-substitute-argument arg checker))
                   (flycheck-checker-arguments checker))))
 
+(defun flycheck--process-send-buffer-contents-chunked (process)
+  "Send contents of current buffer to PROCESS in small batches.
+
+Send the entire buffer to the standard input of PROCESS in chunks
+of 4096 characters.  Chunking is done in Emacs Lisp, hence this
+function is probably far less efficient than
+`send-process-region'.  Use only when required."
+  (let ((from (point-min)))
+    (while (< from (point-max))
+      (let ((to (min (+ from 4096) (point-max))))
+        (process-send-region process from to)
+        (setq from to)))))
+
+(defvar flycheck-chunked-process-input
+  ;; See https://github.com/flycheck/flycheck/issues/794 and
+  ;; https://debbugs.gnu.org/cgi/bugreport.cgi?bug=22344
+  (eq system-type 'windows-nt)
```

Comment:
@cpitclaudel Following [Eli's commit](http://git.savannah.gnu.org/cgit/emacs.git/commit/?h=emacs-25&id=58a622d473112f8ff5b4bdb3e49bc6573dfd3404) that fixed the issue properly we should update the condition to not enable our work around on Emacs 25.1.

The commit appears to introduce a new `w32-pipe-buffer-size` option, which we could use as feature test for this fix, and thus change the condition to `(and (eq system-type 'windows-nt) (not (boundp 'w32-pipe-buffer-size)))`.

What do you think?  

Besides, what about cygwin?  Did any of the reporters of #794 observe the issue on cygwin?